### DISCUSSION TITLE
Ability to insert a module inside/alongside of another one

### DISCUSSION THREAD

**Discussion by iamrasel** [fastfetch-cli/fastfetch#563] [Ideas]
I don't know if it is already available. I couldn't find such thing in the `--help`, so I am guessing it's not.

**Example:**
Custom module can be created like this `--set-keyless CustomModule=Value...Blah`. This is present in the [example 2](https://github.com/fastfetch-cli/fastfetch/raw/dev/screenshots/example2.png). Here a custom module is `--set-keyless HardwareStart=┌───────── \e[1mHardware Information\e[0m ─────────┐`, which outputs `┌───────── Hardware Information ─────────┐`.

**The thing I want to achieve:**
Now, suppose I want to show the output of `Title` module instead of `Hardware Information` string like this `┌───────── user@system ─────────┐`. Or, print `DE` and `WM` in a single line. So, is there already any workaround to achieve this?

**Expected way:**
It would be easier if we could do like below -
`--set-keyless CustomModuleOne=┌───────── $Title ─────────┐`
`--set CustomModuleTwo=$DE, $WM`

Thanks.

**Comment by CarterLi** [comment]
> --set-keyless CustomModuleOne=┌───────── $Title ─────────┐

The closest solution is `-s title --title-format '┌───────── {1}@{2} ─────────┐'`

> --set CustomModuleTwo=$DE, $WM

It's not supported. See https://github.com/fastfetch-cli/fastfetch/issues/482

**Reply by iamrasel** [reply]
> The closest solution is `-s title --title-format '┌───────── {1}@{2} ─────────┐'`

If i'm not wrong, at first I need to add this line, and then add the `Title` module in `--structure`, right? But this doesn't work, it shows `Title` module in default format. I think customizing the `title-format` is not supported like other modules, `fastfetch --help title-format` returns irrelevant information.
> It's not supported. See https://github.com/fastfetch-cli/fastfetch/issues/482

It would be awesome if this could be done.

**Reply by CarterLi** [reply]
It's supported.

<img width="765" alt="image" src="https://github.com/fastfetch-cli/fastfetch/assets/6134068/07750817-1138-4aa3-8330-2ba5304d8dfa">

**Reply by iamrasel** [reply]
Okay, like every other module I added this, `--title-format ┌───── {1}@{2} ─────┐`. It does work (though it's not included in `fastfetch --help title-format`).

But the problem here is, it also shows the _key_ name and _separator_ unlike default state. I tried hiding the _key_ name using `--title-key " "`, but it prints the space.:disappointed: And the _separator_ is still there, so is there any way to hide both of them like default?

**Reply by CarterLi** [reply]
That's why I said `closest` 😆

**Reply by iamrasel** [reply]
Oh, I see.:roll_eyes:

This #482 would be great. And, how about adding `--disable-key` and `--disable-separator`, or something like `--percent-type Int` flag for individual modules?

**Reply by CarterLi** [reply]
See https://github.com/fastfetch-cli/fastfetch/commit/560c2acb888ceb0e5ecbc39f3f382d5e5bfee605

**Reply by CarterLi** [reply]
As for #482, it requires rewriting almost every module and a lot of common code. I dont have enough time and motivation, so no in the near future.

**Reply by iamrasel** [reply]
Okay.

**Comment by iamrasel** [comment]
Another issue (I think) - the default _disk_ format is `{1} / {2} ({3}) - {9}`. So I formatted like this `--disk-format {1} / {2}`, and inserted `--percent-type 2` to show the percentage as bar.

This doesn't work. I mean, when the _disk_ module is formatted, the percentage bar doesn't appear for it.
